gh_pages/_source/session8/Motion-Planning-with-Tesseract.rst

```diff
@@ -0,0 +1,431 @@
+
+   .. code-block:: c++
+
+      auto profile = std::make_shared<tesseract_planning::OMPLDefaultPlanProfile>();
+      profile->planning_time = 10.0;
+  	   profile->planners.reserve(static_cast<std::size_t>(n));
+	   for (Eigen::Index i = 0; i < n; ++i)
+	   {
+	     auto rrt_connect = std::make_shared<tesseract_planning::RRTConnectConfigurator>();
+	     rrt_connect->range = range(i);
+	     profile->planners.push_back(rrt_connect);
+	   }
+	   return profile;
+
+   There are many different OMPL planners available to experiment with. Feel free to play around with a few and observe how your application's motion plan changes (don't forget to include your chosen planner(s) in the header!).
+
+#. Add the planner to the planning server:
+
+   Return to ``planning_server.cpp`` and find the section where we add in our custom planning profiles.
+
+   Copy and paste the follwing
+
+   .. code-block:: c++
+
+      profile_dict_->addProfile<tesseract_planning::OMPLPlanProfile>(
+          tesseract_planning::profile_ns::OMPL_DEFAULT_NAMESPACE, PROFILE, createOMPLProfile());
+
+   Now we have added our new OMPL planning profile to the planning server.
+
+#. Add the planner to the taskflow:
+   
+   Go back to ``taskflow_generators.hpp``. Now we need to include our OMPL profile in our motion planning taskflow. Our OMPL planner will be used with TrajOpt in our freespace taskflow. First, let's create our node for OMPL. Within ``FreespaceMotionPipelineTask`` add
+
+   .. code:: c++
+
+      // Setup OMPL
+      boost::uuids::uuid ompl_planner_task =
+         addNode(std::make_unique<tesseract_planning::OMPLMotionPlannerTask>(output_keys_[0], output_keys_[0]));
+  
+   Now we need to change our graph edges to incorporate these new nodes. Make the following changes to the edges:
+
+   * ``min_length_task`` will connect to ``ompl_planner_task``
+
+   * ``ompl_planner_task`` will connect to ``error_task`` and ``trajopt_planner_task``
+
+   This taskflow now means OMPL will first find planning solutions and then TrajOpt will smooth out the trajectory.
+
+   Now lets return to ``CartesianMotionPipelineTask`` and remove Descartes. Comment out where you created the Descartes node and where you connected ``descartes_planner_task`` to ``error_task`` and ``trajopt_planner_task``. Have ``min_length_task`` connect to ``trajopt_planner_task`` instead of your Descartes node. 
+
+#. Run the application:
+
+   Now try running the full application again with our completed motion planning pipeline. How has the plan changed since step one? Also take a look at our completed taskflow graph again and notice the new taskflow. Try playing around with changing some of the edges and see how the motion plan changes. Here are a few things you could try:
+
+   * Remove TrajOpt and see how Descartes and OMPL perform without it.
+
+   * Remove time parametization
```

Comment:
Removing time param won't have an observable difference in simulation, so I'd probably nix this one.